src/classes/STG_InstallScript.cls

```diff
@@ -51,17 +55,27 @@ global without sharing class STG_InstallScript implements InstallHandler {
             } else if(context.isPush()) {
                 insertTdtmDefaults(null);
             }
+
             UTIL_MasterSchedulableHelper.abortOldScheduledJobs();
+
+            rethrowException = true;
+
+            if (context.previousVersion() == null || context.previousVersion().compareTo(new Version(3, 79)) < 0) {
+                // update record type settings ONLY if upgrading from a version prior to 3.78
+                UTIL_RecordTypeSettingsUpdate.getInstance().updateSettings();
+            }
+
         } catch(Exception e) {
+            Database.rollback(sp);
+
+            if (rethrowException) {
+                throw e;
+                return;
```

Comment:
don't need a return statement after a throw.

Reply:
The return statement has been removed.